File: frontendutilisateur/lib/domain/entities/service_entity.dart
class ServiceEntity {
  final String id;
  final String structureId;
  final String name;
  final String description;
  final double price;
  final String currency;

  ServiceEntity({
    required this.id,
    required this.structureId,
    required this.name,
    required this.description,
    required this.price,
    required this.currency,
  });

  @override
  bool operator ==(Object other) =>
      identical(this, other) ||
      (other is ServiceEntity &&
          runtimeType == other.runtimeType &&
          id == other.id);

  @override
  int get hashCode => id.hashCode;
}

File: src/main/java/com/NND/tech/Structure_Backend/model/entity/ServiceEntity.java
package com.NND.tech.Structure_Backend.model.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "services")
public class ServiceEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;
    
    @Column(nullable = false)
    private String category;

    private String description;
    private BigDecimal price;
    private int duration; // en minutes
    private boolean active = true;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "structure_id", nullable = false)
    private Structure structure;
}
